tools/package-client.js

```diff
@@ -463,6 +566,38 @@ exports.publishPackage = function (options) {
     }
   }
 
+  // Check that our documentation exists (or we know that it doesn't) and has
+  // been filled out.
+  var readmeInfo;
+  buildmessage.enterJob("processing docs", function () {
+    readmeInfo = packageSource.processReadme();
+  });
+  if (buildmessage.jobHasMessages())
+    return;
+  if (readmeInfo && (readmeInfo.hash === files.blankHash)) {
+    buildmessage.error(
+      "Your documentation file is blank, so users may have trouble figuring " +
+      "out how to use your package. Please fill it out, or " +
+      "set 'documentation: null' in your Package.describe");
+  }
+
+  if (readmeInfo &&
+      readmeInfo.excerpt.length > 1500) {
```

Comment:
Yeah, I have mixed feelings about all of them.  Probably the best design (eventually, not needed now) is something where we "preflight" all the metadata with the server before uploading anything?  I guess there's a tradeoff between "should we avoid any network roundtrips if the package doesn't even build" vs "should we avoid a potentially slow compilation if the metadata is blatantly bad".